src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -217,55 +219,37 @@ def login_with_managed_identity(self, identity_id, resource):
 
     def get_user(self, user_or_sp=None):
-        from msal import PublicClientApplication
-        # sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:122
-        from azure.identity._internal.msal_credentials import _load_cache
-        cache = _load_cache()
-        authority = "https://{}/organizations".format(self.authority)
-        app = PublicClientApplication(authority=authority, client_id=_CLIENT_ID, token_cache=cache)
-        return app.get_accounts(user_or_sp)
+        return self._msal_app.get_accounts(user_or_sp)
 
     def logout_user(self, user_or_sp):
-        from msal import PublicClientApplication
-        # sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:122
-        from azure.identity._internal.msal_credentials import _load_cache
-        cache = _load_cache()
-        authority = "https://{}/organizations".format(self.authority)
-        app = PublicClientApplication(authority=authority, client_id=_CLIENT_ID, token_cache=cache)
-
-        accounts = app.get_accounts(user_or_sp)
+        accounts = self._msal_app.get_accounts(user_or_sp)
         logger.info('Before account removal:')
         logger.info(json.dumps(accounts))
 
         for account in accounts:
-            app.remove_account(account)
+            self._msal_app.remove_account(account)
 
-        accounts = app.get_accounts(user_or_sp)
+        accounts = self._msal_app.get_accounts(user_or_sp)
         logger.info('After account removal:')
         logger.info(json.dumps(accounts))
 
     def logout_all(self):
-        from msal import PublicClientApplication
-        from azure.identity._internal.msal_credentials import _load_cache
-        cache = _load_cache()
-        # TODO: Support multi-authority logout
-        authority = "https://{}/organizations".format(self.authority)
-        app = PublicClientApplication(authority=authority, client_id=_CLIENT_ID, token_cache=cache)
-
-        accounts = app.get_accounts()
+        accounts = self._msal_app.get_accounts()
         logger.info('Before account removal:')
         logger.info(json.dumps(accounts))
 
         for account in accounts:
-            app.remove_account(account)
+            self._msal_app.remove_account(account)
 
-        accounts = app.get_accounts()
+        accounts = self._msal_app.get_accounts()
         logger.info('After account removal:')
         logger.info(json.dumps(accounts))
 
     def get_user_credential(self, home_account_id, username):
-        auth_profile = AuthProfile(self.authority, home_account_id, self.tenant_id, username)
-        return InteractiveBrowserCredential(profile=auth_profile, silent_auth_only=True)
+        auth_record = AuthenticationRecord(self.tenant_id, self.client_id, self.authority,
+                                           home_account_id, username)
+        return InteractiveBrowserCredential(authentication_record=auth_record, disable_automatic_authentication=True,
```

Comment:
`InteractiveCredential` which is the base class of `InteractiveBrowserCredential` and `DeviceCodeCredential` should be exposed in `azure/identity/__init__.py`.

CLI won't know which type the user used to login before. Choosing any one of `InteractiveBrowserCredential` and `DeviceCodeCredential` works, yes, but this is not a good programming practice as we are using the base class from sub class. See [Teams thread: InteractiveCredential should be exposed](https://teams.microsoft.com/l/message/19:85462ed2792e424bb248555e238caa8c@thread.skype/1589357216410?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1589357216410&teamName=Azure%20SDK&channelName=Service%20-%20Azure.Identity&createdTime=1589357216410)

![image](https://user-images.githubusercontent.com/4003950/81794777-de19d500-953d-11ea-8b29-df67431f1555.png)